Core.ApplicationServices/ItSystemUsageMigration/ItSystemUsageMigrationService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Core.ApplicationServices.Authorization;
+using Core.DomainModel;
+using Core.DomainModel.ItSystem;
+using Core.DomainServices;
+using Core.DomainServices.Extensions;
+
+namespace Core.ApplicationServices.ItSystemUsageMigration
+{
+    public class ItSystemUsageMigrationService : IItSystemUsageMigrationService
+    {
+        private IAuthorizationContext _authorizationContext;
+        private readonly IGenericRepository<ItSystem> _itSystemRepository;
+
+        public ItSystemUsageMigrationService(IAuthorizationContext authorizationContext, IGenericRepository<ItSystem> itSystemRepository)
+        {
+            _authorizationContext = authorizationContext;
+            _itSystemRepository = itSystemRepository;
+        }
+
+        public Result<ResultStatus, IEnumerable<ItSystem>> GetUnusedItSystemsByOrganization(int organizationId, string nameContent, int numberOfItSystems, bool getPublic)
+        {
+
+            if (getPublic)
+            {
+                var itSystems = _itSystemRepository
+                    .AsQueryable()
+                    .ByPublicAccessOrOrganizationId(organizationId)
```

Comment:
ikke to næsten identiske blokke. Den udfører ikke query fordi du bryder den op og det er jo kun ByOrganizationId vs. ByPublicAccessOrORganizationId du vil skifte imellem. 

så:

```
var itsystems = _itSystemsRepository.AsQueryable();
itSystems = getPublic ? itSystems.ByPublicAccessOrOrganizationId(organizationId) : ByOrganizationId(organizationId);
itSystems = itSystems.
                .ExceptByInUsage(organizationId)
                    .ByPartOfName(nameContent)
                    .OrderBy(x => x.Name)
                    .Take(numberOfItSystems);
```
Indtil du trækker elementer ud opbygger den bare query

Reply:
Og slut af med toList()